### PR TITLE
Next batch of tests

### PR DESCRIPTION
**Pull request by rossberg-chromium**
This time:
- loop
- br
- return
- bunch of additions to previous tests

Plus a bug fix in the pretty printer.

### CODE REVIEW

File: ml-proto/test/block.wast [hunk 1]
```diff
@@ -46,43 +46,62 @@
     (f32.gt (block (call $nop) (f32.const 3)) (block (call $nop) (f32.const 3)))
   )
 
-  (func "br-bare" (result i32)
+  (func "break-bare" (result i32)
```
Comment (sunfishcode) [inline ml-proto/test/block.wast]: Please do not rename "br" to "break" in the tests; the instruction is named "br".

Comment (rossberg-chromium) [inline ml-proto/test/block.wast]: This is to match the loop test cases, where the names distinguish between breaking and continuing (see below). So the naming does not refer to the operator, but to the control flow transfer happening.

Comment (sunfishcode) [inline ml-proto/test/block.wast]: Ok. I guess I'm already in the habit of thinking in terms of https://github.com/WebAssembly/design/pull/652. If we do that, we can simplify these tests.

**Comment by rossberg-chromium** [comment]
Ping :)

**Comment by rossberg-chromium** [comment]
Any opposition to landing these tests?

**Comment by lukewagner** [comment]
lgtm

**Pull request merged**